### PR TITLE
Add backstage incremental ingestion backend plugin

### PR DESCRIPTION
**Pull request by wKich**
## Motivation

We want to make the Incremental Ingestion Backend plugin available to Backstage community. 

This is the first step in doing this.

## Approach

Moved it to separate backstage plugin

### CODE REVIEW

File: plugins/incremental-ingestion-backend/migrations/20220613125155_init.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/* eslint-disable func-names */
+exports.up = async function (knex) {
+  await knex.raw(`
+CREATE VIEW ingestion.current_entities as SELECT
+  FORMAT('%s:%s/%s',
+    LOWER(final_entity::json #>> '{kind}'),
+    LOWER(final_entity::json #>> '{metadata, namespace}'),
+    LOWER(final_entity::json #>> '{metadata, name}')) as ref,
+  final_entity::json #>> '{metadata, annotations, hp.com/provider-name}' as provider_name,
```
Comment (taras) [inline plugins/incremental-ingestion-backend/migrations/20220613125155_init.js]: We should make this generic and/or configurable.

**Comment by frontsidejack** [comment]
The following preview packages were published:
|                                          Name                                         |                            Version                           | Install Command                                                                                                         |
| :-----------------------------------------------------------------------------------: | :----------------------------------------------------------: | ----------------------------------------------------------------------------------------------------------------------- |
| <br><pre><code>@frontside/backstage-plugin-incremental-ingestion-backend</code></pre> | <br><pre><code>0.1.1-dl-incremental-ingestion.1</code></pre> | <br><pre><code>yarn add @frontside/backstage-plugin-incremental-ingestion-backend@dl-incremental-ingestion</code></pre> |
<p align='right'><em>Generated by <a href='https://github.com/thefrontside/actions/tree/main/publish-pr-preview'>@thefrontside/actions Frontside</a><em></p>

File: plugins/incremental-ingestion-backend/migrations/20220613125155_init.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/* eslint-disable func-names */
+exports.up = async function (knex) {
+  await knex.raw(`
+CREATE VIEW ingestion.current_entities as SELECT
+  FORMAT('%s:%s/%s',
+    LOWER(final_entity::json #>> '{kind}'),
+    LOWER(final_entity::json #>> '{metadata, namespace}'),
+    LOWER(final_entity::json #>> '{metadata, name}')) as ref,
```
Comment (taras) [inline plugins/incremental-ingestion-backend/migrations/20220613125155_init.js]: Can we use `refresh_state` table as suggested for our batch loader?

File: plugins/incremental-ingestion-backend/src/iteration-db.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    const added = entities.map(deferred => ({
+      ...deferred,
+      entity: {
+        ...deferred.entity,
+        metadata: {
+          ...deferred.entity.metadata,
+          annotations: {
+            ...deferred.entity.metadata.annotations,
+            'hp.com/provider-name': provider.getProviderName(),
```
Comment (taras) [inline plugins/incremental-ingestion-backend/src/iteration-db.ts]: We should make this configurable and default to `@frontside.com/incremental-provider-name`

File: plugins/incremental-ingestion-backend/migrations/20220613125155_init.js (hunk 1 above)
Comment (taras) [inline plugins/incremental-ingestion-backend/migrations/20220613125155_init.js]: We should make this configurable and default to `@frontside.com/incremental-provider-name`

**Comment by wKich** [comment]
@taras done. Replaced view to index and make annotation field customizable

**Review by taras — APPROVED** [review]
👍

**Pull request merged**